### ISSUE TITLE
How can I use multiple SSM networks for speculative inference at the same time, please?

### ISSUE THREAD

**Issue by zym1599** [flexflow/FlexFlow#1262]
How can I use multiple SSM networks for speculative inference at the same time, please? Also, why is the inference time longer after using the SSM network? Is this the reasoning time on the chart? Or does it include the data loading time?
![image](https://github.com/flexflow/FlexFlow/assets/42165621/08d6d61d-569e-42a1-a54a-8841482684f9)

thanks alot！

**Comment by RaghavvGoel** [comment]
Hi, following up on this, is there a script which allows me to use multiple SSMs? 
The example in READMDE only shows for a single SSM. 
Also where can I find the multiple SSMs, the weight folders given only have a single SSM model

Thanks a lot for the help

**Comment by xinlong-yang** [comment]
> Hi, following up on this, is there a script which allows me to use multiple SSMs? The example in READMDE only shows for a single SSM. Also where can I find the multiple SSMs, the weight folders given only have a single SSM model
> 
> Thanks a lot for the help

hello, have you solved this?

**Comment by dutsc** [comment]
`./inference/spec_infer/spec_infer \
    -ll:gpu 4 \
    -ll:fsize 22000 \
    -ll:zsize 30000 \
    -llm-model /models/opt-13b/ \
    -ssm-model /models/opt-125m/ \
    -ssm-model /models/opt-125m/ \
    -ssm-model /models/opt-125m/ \
    -prompt /workspace/FlexFlow/prompts/prompt.json \
    -tensor-parallelism-degree 4 \
    --fusion > ../sclog/spec_infer.log`
   
   This is my script to run multiple SSMs with cpp interface. I hope it helps.

**Comment by RaghavvGoel** [comment]
Thanks a lot! 
Does this mean the same ssm model is loaded 4 times?

**Comment by dutsc** [comment]
Yes, the load will be repeated as many times as you use the -ssm-model parameter.

**Comment by RaghavvGoel** [comment]
Got it, thanks a lot for answering! 

As a side note, for some reason I thought SpecInfer uses 4 different SSM models of same size through their SSM training pipeline, where after training the first SSM, the second SSM (same size) will be trained on all the samples which the first SSM got wrong and so on. So I thought this would lead to different weights of SSM for each step. 
I might have missed something in the paper and will re-check :)
